app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdUpdate.kt

```diff
@@ -57,17 +57,20 @@ class ScanSimpleIdUpdate @Inject constructor(
 
   private fun patientFoundByIdentifierSearch(patients: List<Patient>, identifier: Identifier): ScanSimpleIdEffect {
     return if (patients.size > 1) {
-      try {
-        OpenShortCodeSearch(BpPassport.shortCode(identifier))
-      } catch (e: Exception) {
-        OpenShortCodeSearch(identifier.value)
-      }
+      multiplePatientsWithId(identifier)
     } else {
       val patientId = patients.first().uuid
       OpenPatientSummary(patientId)
     }
   }
 
+  private fun multiplePatientsWithId(identifier: Identifier): ScanSimpleIdEffect {
+    return when (identifier.type) {
+      BpPassport -> OpenShortCodeSearch(BpPassport.shortCode(identifier))
+      else -> OpenPatientSearch(null, initialSearchQuery = identifier.value)
```

Comment:
Why are we passing the identifier as null here and in the tests as well? Does this mean apart from BP passport just search with ID regardless of whatever identifier it is?

Reply:
Yep, if we pass an identifier to instant search it tries to bring up the sheet that asks if we want to register or assign it to a patient. In this case, this is not a blank identifier, we know this is already assigned to a patient. I should probably switch to using factory constructors for this to make it much more apparent what it's doing 🤔